linkedlist.c

```diff
@@ -0,0 +1,61 @@
+#include<stdio.h>
+#include<stdlib.h>
+
+struct node
+{
+int data;
+struct node *next;
+};
+
+struct node *head=NULL;
+
+
+void insertion_beginning()
+{
+struct node *newNode;
+newNode =NULL;
+newNode=(struct node*)malloc(sizeof(struct node));
+printf("enter data :");
+scanf("%d",&newNode->data);
+if(head==NULL)
+{
+head=newNode;
+}
+else
+{
+newNode->next=head;
+head=newNode;
+}
+printf("\nNode inserted\n");
+}
+
+void display()
+{
+int count=0;
+struct node *newNode;
+printf("Available data:\n");
+newNode=head;
+while(newNode!=NULL)

```

Comment:
This will go on for a infinite loop printing numbers.
[**Hint:** You are missing a initialization somewhere outside this function. That will solve this.]

Reply:
No, it's working fine it will print untill newnode != NULL . Once it's reach the newnode ==NULL it's stop printing.

Reply:
ok I will add